src/Avalonia.Controls/Notifications/WindowNotificationManager.cs

```diff
@@ -135,20 +186,36 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
             {
                 UpdatePseudoClasses(change.GetNewValue<NotificationPosition>());
             }
+
+            if (change.Property == HostProperty)
+            {
+                Install();
+            }
         }
 
         /// <summary>
         /// Installs the <see cref="WindowNotificationManager"/> within the <see cref="AdornerLayer"/>
-        /// of the host <see cref="Window"/>.
         /// </summary>
-        /// <param name="host">The <see cref="Window"/> that will be the host.</param>
-        private void Install(TemplatedControl host)
+        private void Install()
         {
-            var adornerLayer = host.FindDescendantOfType<VisualLayerManager>()?.AdornerLayer;
+            // unregister from AdornerLayer if this control was already installed
+            if (adornerLayer is not null && !adornerLayer.Children.Contains(this))
+            {
+                adornerLayer.Children.Remove(this);
+            }
+            
+            // Try to get the host. If host was null, use the TopLevel instead.
+            var host = Host ?? Parent as Visual;
+
+            if (host is null) throw new InvalidOperationException("NotificationControl cannot be installed. Host was not found.");
+
+            adornerLayer = host is TopLevel 
+                ? host.FindDescendantOfType<VisualLayerManager>()?.AdornerLayer 
+                : AdornerLayer.GetAdornerLayer(host);
 
             if (adornerLayer is not null)
             {
-                adornerLayer.Children.Add(this);
+                // adornerLayer.Children.Add(this);
```

Comment:
Why is this line commented? Especially since `adornerLayer.Children.Remove` is still there.